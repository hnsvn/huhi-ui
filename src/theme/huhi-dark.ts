import ITheme from './theme-interface'
import colors from './colors'
import defaultTheme from './huhi-default'

const darkTheme: ITheme = {
  ...defaultTheme,
  name: 'Huhi Dark',
  color: {
    ...defaultTheme.color,
    // brand
    interactive07: colors.grey000,
    interactive08: colors.grey700,
    // regular controls
    defaultControl: colors.grey400,
    defaultControlInteracting: colors.white,
    defaultControlActive: colors.grey500,
    // text
    contextMenuForeground: colors.white,
    contextMenuHoverForeground: colors.white,
    text: colors.white,
    text01: colors.grey000,
    text02: colors.grey400,
    text03: colors.grey600,
    // backgrounds
    contextMenuBackground: colors.black,
    contextMenuHoverBackground: colors.blurple500,
    panelBackground: colors.grey900,
    background01: '#17171F',
    background02: colors.grey900,
    // dividers and outlines
    divider01: colors.grey800,
    // alerts
    errorText: '#FF4B6A',
    errorBackground: 'rgba(227, 36, 68, 0.2)',
    warningBackground: 'rgba(255, 221, 99, 0.2)',
    successBackground: 'rgba(81, 207, 102, 0.3)'
  }
}

export default darkTheme
