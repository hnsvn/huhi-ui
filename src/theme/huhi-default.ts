import colors from './colors'
import ITheme from './theme-interface'

// Define colors, fonts, and sizes by purpose.
// Keys should not describe the value, but what they are to be used for.

const theme: ITheme = {
  name: 'Default',
  palette: {
    ...colors
  },
  color: {
    // brand
    interactive01: '#EA3A0D',
    interactive02: colors.orange500,
    interactive03: colors.orange400,
    interactive04: colors.blurple600,
    interactive05: colors.blurple500,
    interactive06: colors.blurple400,
    interactive07: colors.neutral900,
    interactive08: colors.grey500,
    focusBorder: colors.blurple300,
    brandHuhi: colors.orange500,
    brandBat: colors.blurple500,
    // huhi controls
    brandHuhiInteracting: colors.orange500,
    brandHuhiActive: colors.orange200,
    brandHuhiLight: colors.orange000,
    // bat controls
    brandBatInteracting: colors.blurple500,
    brandBatActive: colors.blurple200,
    // regular controls
    defaultControl: colors.grey800,
    defaultControlInteracting: colors.black,
    defaultControlActive: colors.grey700,
    // warning controls
    warn: colors.red500,
    warnInteracting: colors.red600,
    warnActive: colors.red700,
    // subtle controls
    subtle: colors.grey400,
    subtleBackground: colors.grey100,
    subtleExclude: colors.grey300,
    subtleInteracting: colors.grey500,
    subtleActive: colors.grey600,
    // disabled controls
    disabled: colors.grey300,
    disabledResourceBlocked: colors.neutral600,
    // backgrounds
    panelBackground: colors.white,
    panelBackgroundSecondary: colors.neutral000,
    primaryBackground: colors.white,
    contextMenuBackground: colors.white,
    contextMenuHoverBackground: colors.blurple200,
    secondaryBackground: colors.grey400,
    modalOverlayBackground: 'rgba(36,37,54,0.85)',
    background01: colors.neutral000,
    background02: colors.white,
    background03: colors.grey800,
    // text
    contextMenuForeground: colors.grey800,
    contextMenuHoverForeground: colors.neutral800,
    detailDescription: colors.grey500,
    text: colors.grey700,
    text01: colors.neutral900,
    text02: colors.neutral700,
    text03: colors.neutral600,
    // form controls
    inputBorder: colors.grey500,
    separatorLine: colors.grey100,
    // images
    lionLogo: colors.grey500,
    // dividers and outlines
    divider01: '#E9E9F4',
    // alerts
    errorText: colors.red600,
    errorIcon: colors.red600,
    errorBorder: colors.red500,
    errorBackground: colors.red000,
    warningIcon: colors.yellow500,
    warningBorder: colors.yellow400,
    warningBackground: colors.yellow000,
    infoIcon: colors.blue500,
    infoBorder: colors.blue400,
    infoBackground: 'rgba(93, 181, 252, 0.2)',
    successIcon: colors.teal600,
    successBorder: colors.green500,
    successBackground: colors.green200
  },
  fontFamily: {
    heading: 'Poppins, sans-serif',
    body: 'Muli, sans-serif',
    system: '-apple-system,system-ui,BlinkMacSystemFont,"Segoe UI",Roboto,"Helvetica Neue",Arial,sans-serif'
  }
}

export default theme
