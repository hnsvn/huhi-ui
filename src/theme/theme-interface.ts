import { ThemedStyledProps } from 'styled-components'

export default interface IThemeProps {
  name: string,
  palette: { [key: string]: string }
  color: {
    contextMenuBackground: string
    contextMenuForeground: string
    contextMenuHoverBackground: string
    contextMenuHoverForeground: string
    background01: string
    background02: string
    background03: string
    brandHuhi: string
    brandHuhiInteracting: string
    brandHuhiActive: string
    brandHuhiLight: string
    brandBat: string
    brandBatInteracting: string
    brandBatActive: string
    detailDescription: string
    disabled: string
    disabledResourceBlocked: string
    primaryBackground: string
    secondaryBackground: string
    modalOverlayBackground: string
    defaultControl: string
    defaultControlInteracting: string
    defaultControlActive: string
    warn: string
    warnInteracting: string
    warnActive: string
    subtle: string
    subtleBackground: string
    subtleExclude: string
    subtleInteracting: string
    subtleActive: string
    text: string
    text01: string
    text02: string
    text03: string
    panelBackground: string
    panelBackgroundSecondary: string
    lionLogo: string
    inputBorder: string
    separatorLine: string
    interactive01: string
    interactive02: string
    interactive03: string
    interactive04: string
    interactive05: string
    interactive06: string
    interactive07: string
    interactive08: string
    focusBorder: string
    divider01: string
    errorIcon: string
    errorText: string
    errorBorder: string
    errorBackground: string
    warningIcon: string
    warningBorder: string
    warningBackground: string
    infoIcon: string
    infoBorder: string
    infoBackground: string
    successIcon: string
    successBorder: string
    successBackground: string

  }
  fontFamily: {
    heading: string
    body: string
    system: string
  }
}

export type HuhiThemedStyledProps<T> = ThemedStyledProps<T, IThemeProps>
