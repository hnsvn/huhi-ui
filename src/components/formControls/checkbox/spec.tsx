/* global jest, expect, describe, it, afterEach */
import * as React from 'react'
import { create } from 'react-test-renderer'
import '@testing-library/jest-dom'
import userEvent from '@testing-library/user-event'
import { render, screen, getByTestId } from '@testing-library/react'
import Checkbox, { Props } from './index'
import { TestThemeProvider } from '../../../theme'

function TestCheckbox(props: {id: string}) {
  return (
    <Checkbox
      value={{ 'yt': true, 'tw': false, 'inst': false }}
      testId={props.id}
      multiple={false}
    >
      <div data-key='yt'>YouTube</div>
      <div data-key='tw'>Twitter</div>
      <div data-key='inst'>Instagram</div>
    </Checkbox>
  )
}

function getCheckboxElementByTestId (container: HTMLElement, testId: string) {
  // Can't use getByTestId since Checkbox uses data-test-id instead of data-testid
  return container.querySelector<HTMLElement>(`[data-test-id=${testId}]`)
}

describe('Checkbox tests', () => {
  const baseComponent = (props?: Props) =>
    <TestThemeProvider>
      <TestCheckbox id={'checkbox'} />
    </TestThemeProvider>

  describe('basic tests', () => {
    it('matches the snapshot', () => {
      const component = baseComponent()
      const tree = create(component).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('renders the component', () => {
      const test = render(baseComponent())
      const element = getCheckboxElementByTestId(test.container, 'checkbox')
      expect(element).toBeInTheDocument()
    })

    it('is focusable', () => {
      const testContent = (
        <TestThemeProvider>
          <Checkbox value={{a: true}} testId={'checkbox1'}><span data-key="a">first</span></Checkbox>
          <Checkbox value={{a: true}} testId={'checkbox2'}><span data-key="a">second</span></Checkbox>
        </TestThemeProvider>
      )
      const test = render(testContent)
      const first = getCheckboxElementByTestId(test.container, 'checkbox1')
      const second = getCheckboxElementByTestId(test.container, 'checkbox2')
      expect(first).toBeTruthy()
      expect(second).toBeTruthy()
      const firstCheck = getByTestId(first, 'checkbox-child-0')
      const secondCheck = getByTestId(second, 'checkbox-child-0')
      firstCheck.focus()
      expect(firstCheck).toHaveFocus()
      userEvent.tab()
      expect(secondCheck).toHaveFocus()
    })

    it('can be used by keyboard', () => {
      const onClickFirst = jest.fn()
      const onClickSecond = jest.fn()

      const testContent = (
        <TestThemeProvider>
          <Checkbox value={{a: true}} testId={'checkbox1'} onChange={onClickFirst}><span data-key="a">first</span></Checkbox>
          <Checkbox value={{a: true}} testId={'checkbox2'} onChange={onClickSecond}><span data-key="a">second</span></Checkbox>
        </TestThemeProvider>
      )
      const test = render(testContent)
      // Can't use getByTestId since Checkbox uses data-test-id instead of data-testid
      const first = getCheckboxElementByTestId(test.container, 'checkbox1')
      const second = getCheckboxElementByTestId(test.container, 'checkbox2')
      expect(first).toBeTruthy()
      expect(second).toBeTruthy()
      const firstCheck = getByTestId(first, 'checkbox-child-0')
      const secondCheck = getByTestId(second, 'checkbox-child-0')
      expect(onClickFirst).toHaveBeenCalledTimes(0)
      firstCheck.focus()
      // Note(petemill): Using [enter] for all cases as cannot get space to work via ' ' or [space]
      userEvent.keyboard('[enter]')
      expect(onClickFirst).toHaveBeenCalledTimes(1)
      userEvent.tab()
      expect(onClickSecond).toHaveBeenCalledTimes(0)
      userEvent.keyboard('[enter]')
      expect(onClickSecond).toHaveBeenCalledTimes(1)
      expect(onClickFirst).toHaveBeenCalledTimes(1)
    })
  })
})
