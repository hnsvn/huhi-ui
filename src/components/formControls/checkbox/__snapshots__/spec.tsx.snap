// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox tests basic tests matches the snapshot 1`] = `
.c0 {
  --checkbox-border-color: #A1A8F2;
  --checkbox-border-size: 1px;
  --checkbox-box-size: 18px;
  --checkbox-box-color: #696FDC;
  --checkbox-box-spacing: 12px;
  --checkbox-label-color: #686978;
  --checkbox-label-size: 14px;
  font-family: Muli,sans-serif;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 4px;
  color: var(--checkbox-label-color);
  font-size: var(--checkbox-label-size);
  cursor: pointer;
}

.c0:focus,
.c0:hover:not([disabled]) {
  outline: none;
  --checkbox-border-color: #A0A5EB;
  --checkbox-border-size: 3px;
}

.c0:focus {
  outline: none;
}

.c4 {
  --checkbox-border-color: #D1D1DB;
  --checkbox-border-size: 1px;
  --checkbox-box-size: 18px;
  --checkbox-box-color: #696FDC;
  --checkbox-box-spacing: 12px;
  --checkbox-label-color: #686978;
  --checkbox-label-size: 14px;
  font-family: Muli,sans-serif;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 4px;
  color: var(--checkbox-label-color);
  font-size: var(--checkbox-label-size);
  cursor: pointer;
}

.c4:focus,
.c4:hover:not([disabled]) {
  outline: none;
  --checkbox-border-color: #A0A5EB;
  --checkbox-border-size: 3px;
}

.c4:focus {
  outline: none;
}

.c1 {
  border-radius: 4px;
  position: relative;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 3px;
  -webkit-flex-basis: var(--checkbox-box-size);
  -ms-flex-preferred-size: var(--checkbox-box-size);
  flex-basis: var(--checkbox-box-size);
  width: var(--checkbox-box-size);
  height: var(--checkbox-box-size);
  color: var(--checkbox-box-color);
  margin: 8px;
}

.c1:after {
  -webkit-transition: border .1s ease-in-out;
  transition: border .1s ease-in-out;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  border: var(--checkbox-border-size) solid var(--checkbox-border-color);
  content: '';
  display: block;
  border-radius: 4px;
}

.c3 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding-top: 1px;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  width: 100%;
  height: 100%;
  fill: currentColor;
}

<div
  data-test-id="checkbox"
>
  <label
    aria-checked="true"
    className="c0"
    data-testid="checkbox-child-0"
    disabled={false}
    onClick={[Function]}
    onKeyPress={[Function]}
    role="checkbox"
    selected={true}
    size="small"
    tabIndex={0}
    type="light"
  >
    <span
      className="c1"
    >
      <svg
        aria-hidden="true"
        className="c2"
        focusable="false"
        viewBox="0 0 32 32"
      >
        <path
          d="M13 26a1 1 0 0 1-.67-.26l-9-8.2a1 1 0 0 1 1.34-1.48l8.23 7.49L27.23 6.13A1 1 0 0 1 28.64 6a1 1 0 0 1 .13 1.4l-15 18.24a1 1 0 0 1-.7.36z"
        />
      </svg>
    </span>
    <span
      className="c3"
      disabled={false}
      size="small"
    >
      YouTube
    </span>
  </label>
  <label
    aria-checked="false"
    className="c4"
    data-testid="checkbox-child-1"
    disabled={false}
    onClick={[Function]}
    onKeyPress={[Function]}
    role="checkbox"
    selected={false}
    size="small"
    tabIndex={0}
    type="light"
  >
    <span
      className="c1"
    />
    <span
      className="c3"
      disabled={false}
      size="small"
    >
      Twitter
    </span>
  </label>
  <label
    aria-checked="false"
    className="c4"
    data-testid="checkbox-child-2"
    disabled={false}
    onClick={[Function]}
    onKeyPress={[Function]}
    role="checkbox"
    selected={false}
    size="small"
    tabIndex={0}
    type="light"
  >
    <span
      className="c1"
    />
    <span
      className="c3"
      disabled={false}
      size="small"
    >
      Instagram
    </span>
  </label>
</div>
`;
